### PR TITLE
[#9] add codecov coverage reporting

### LINKED ISSUE #9: Setup test coverage CI job
- [ ] integrate https://codecov.io into the `test` job in `.gitlab-ci.yml` (the existing coverage setting there doesn't really do what we expect)

Cosmos does it using:
- https://github.com/cosmos/cosmos-sdk/blob/develop/.codecov.yml
- https://github.com/cosmos/cosmos-sdk/blob/develop/.circleci/config.yml#L245

### PR DESCRIPTION
**Pull request by cpolitano**
Closes #9

### CODE REVIEW

**Comment by aaronc** [comment]
I see there were errors generating the coverage report: https://codecov.io/github/regen-network/regen-ledger/commit/a0724a6f348fc42dcde42b55db1af65e591a23de

Also, there's a way to get the codecov bot to place a report here in the PR. See any of the PR's in cosmos-sdk. Let's see if we can get that working first

File: .codecov.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    changes: false
+
+comment:
+  layout: "header, diff"
+  behavior: default  # update if exists else create new
+
+ignore:
+  - "docs"
+  - "*.md"
+  - "*.rst"
```
[the remark this answers is not in the document]
Comment (cpolitano) [reply .codecov.yml]: I didn't see a way to do that in their docs.

**Comment by aaronc** [comment]
Is there anyway to get Codecov to report coverage for all packages? Take a look at this: https://codecov.io/gh/regen-network/regen-ledger/pull/12/tree?path=x. It only shows coverage info for packages that actually have tests, erroneously indicating that we have 41% coverage when in fact it is much lower because most packages have no tests at all.

**Comment by cpolitano** [comment]
I think we'd have to add test files for any module we want included in the coverage report.

**Comment by aaronc** [comment]
Okay, let's merge this. It would be good to setup stubs at some point wherever we want tests so that we get realistic metrics.

**Pull request merged**